lib/holdings/item.rb

```diff
@@ -134,8 +133,17 @@ def stackmapable?
       stackmapable_library? && stackmapable_location?
     end
 
+    def bound_with_parent
+      return nil unless document && document['holdings_json_struct']
+
+      match = document['holdings_json_struct'].first['holdings']&.find do |item|
+        item.dig('boundWith', 'item', 'barcode') == barcode
```

Comment:
```suggestion
        item.dig('boundWith', 'item', 'id') == id
```

I think we've seen items without barcodes or duplicated barcodes (😬). `id` ought to always be available though.